specs/validator-system-v2/asset-manager.md

```diff
@@ -0,0 +1,223 @@
+
+## Composition of Asset Manager
+
+Asset Manager incorporates a `Vault` component, inspired by the ERC-4626 standard. Its responsibilities include:
+
+1. Management of delegated KRO and KGH NFTs
+
+    - Adopts the structure of ERC-4626 to issue shares when KRO and KGH NFTs are delegated.
+
+2. Slashing and Reward Calculation
+
+    - Calculates rewards for delegators based on the output root submission and manages slashing of assets when
+   challenges occur.
+
+Each validator registered in `ValidatorManager` has a corresponding `Vault` struct. The `Vault` struct is defined as
+follows:
+
+```solidity
+struct Vault {
+    Asset asset;
+    Pending pending;
+    mapping(address => KroDelegator) kroDelegators;
+    mapping(address => KghDelegator) kghDelegators;
+}
+```
+
+The `Asset` and `Pending` structs contain global information such as the total number of KROs in the `Vault`. In
+contrast, the `KroDelegator` and `KghDelegator` structs store individual information about each delegator,
+such as the number of shares he hold in the `Vault`.
+
+The `Vault` defines separate assets and shares for KRO and KGH NFTs, allowing for distinct reward distribution
+mechanisms for KRO delegators and KGH NFT delegators.
+
+|       | asset                                                                   | share                                                        |
+|-------|-------------------------------------------------------------------------|--------------------------------------------------------------|
+| `KRO` | KRO deposited by delegators & validators + Cumulative base reward       | Shares from KRO delegation and KRO inside delegated KGH NFTs |
+| `KGH` | Virtual KRO generated by KGH NFT delegation + Cumulative boosted reward | Shares from KGH NFT delegation                               |
+
+Consequently, there are two types of shares that a delegator can receive: `kroShare` and `kghShare`. For example,
+if a delegator deposits KRO, they receive `kroShare` based on the current state of the `Vault`. The number of shares
+issued is calculated similarly to ERC-4626, as follows:
+
+$$amount \times \frac{totalKroShares+10^{offset}}{totalKroAssets+1}$$
+
+If a delegator wants to undelegate all their assets by burning `kroShare`, the amount of KRO they receive is calculated
+as follows:
+
+$$amount \times \frac{totalKroAssets+1}{totalKroShares+10^{offset}}$$
+
+When KGH NFT is delegated, it is considered as delegating an amount of KRO equal to `VKRO_PER_KGH` per KGH NFT, and
+corresponding shares are issued.
+
+If the same amount of shares is issued for each KGH NFT, then holders who delegate later would gain a share of the
+previously accumulated boosted rewards. To prevent this, when KGH NFT is delegated, it is converted to a fixed amount
+of virtual KRO, which is then used to calculate the shares that the user should receive.
+
+When a user delegates 1 KGH, an amount of virtual KRO equal to `VKRO_PER_KGH` is generated.
```

Comment:
`generated` might be confusing. I think this paragraph can be removed. Above description is enough